src/Okta.Sdk.IntegrationTests/PoliciesScenarios.cs

```diff
@@ -0,0 +1,1197 @@
+                ((IOktaSignOnPolicyRule)updatedPolicyRule).Conditions.Network.Connection.Should().Be("ANYWHERE");
+            }
+            finally
+            {
+                await client.Policies.DeactivatePolicyRuleAsync(createdPolicy.Id, createdPolicyRule.Id);
+                await client.Policies.DeletePolicyRuleAsync(createdPolicy.Id, createdPolicyRule.Id);
+                await client.Policies.DeactivatePolicyAsync(createdPolicy.Id);
+                await client.Policies.DeletePolicyAsync(createdPolicy.Id);
+            }
+        }
+
+        [Fact]
+        public async Task GetPolicyRules()
+        {
+            var client = TestClient.Create();
+            var guid = Guid.NewGuid();
+
+            var policy = new PasswordPolicy()
+            {
+                // Name has a maximum of 50 chars
+                Name = $"dotnet-sdk: GetPolicyRules {guid}".Substring(0, 50),
+                Type = PolicyType.Password,
+                Status = "ACTIVE",
+                Description = "The default policy applies in all situations if no other policy applies.",
+            };
+
+            var createdPolicy = await client.Policies.CreatePolicyAsync(policy);
+            // NIT: Create an AddSignOnPolicyRuleOptions helper class with plain properties, encapsulate all the creation logic below and use this instead: client.Policies.AddPolicyRuleAsync(addSignOnPolicyRuleOptions);
+            IPasswordPolicyRule policyRule = new PasswordPolicyRule()
+            {
+                Name = $"dotnet-sdk: GetPolicyRules {guid}".Substring(0, 50),
+                Type = PolicyType.Password,
+                Conditions = new PasswordPolicyRuleConditions()
+                {
+                    People = new PolicyPeopleCondition()
+                    {
+                        Users = new UserCondition()
+                        {
+                            Exclude = new List<string>(),
+                        },
+                    },
+                    Network = new PolicyNetworkCondition()
+                    {
+                        Connection = "ANYWHERE",
+                    },
+                },
+                Actions = new PasswordPolicyRuleActions()
+                {
+                    PasswordChange = new PasswordPolicyRuleAction()
+                    {
+                        Access = "ALLOW",
+                    },
+                    SelfServicePasswordReset = new PasswordPolicyRuleAction()
+                    {
+                        Access = "ALLOW",
+                    },
+                    SelfServiceUnlock = new PasswordPolicyRuleAction()
+                    {
+                        Access = "DENY",
+                    },
+                },
+            };
+
+            var createdPolicyRule = await client.Policies.AddPolicyRuleAsync(policyRule, createdPolicy.Id);
+
+            try
+            {
+                var policyRules = await createdPolicy.ListPolicyRules().ToList();
```

Comment:
Can I use `OfType<IPasswordPolicyRule>` as well?

Reply:
@nbarbettini yes, we have an IT that covers that case. Take a look at `GetPolicyOfType()`:
```
 var retrievedPolicy = await client.Policies.GetPolicyAsync<IOktaSignOnPolicy>(createdPolicy.Id);

retrievedPolicy.Should().NotBeNull();
                retrievedPolicy.Should().BeAssignableTo<IOktaSignOnPolicy>();
```